pyproject.toml

```diff
@@ -0,0 +1,92 @@
+[build-system]
+requires = [
+    "hatchling>=1.13.0",
+    "hatch-vcs>=0.3.0",
+]
+build-backend = "hatchling.build"
+
+[project]
+name = "pyshaper"
+dynamic = ["version"]
+description = "framework for defining, building, and evaluating generalized shape observables for collider physics"
+readme = "README.md"
+license = { text = "MIT" }  # SPDX short identifier
+requires-python = ">=3.7"
+authors = [
+    { name = "Rikab Gambhir", email = "rikab@mit.edu" },
+]
+keywords = [
+    "shape observables",
+    "jet physics",
+]
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Physics",
+]
+dependencies = [
+    "torch>=1.10.0",
+    "geomloss>=0.2.3",
+    "pyjet>=1.9.0",  # FIXME: Deprecated
+    "scipy>=1.5.1",
+    "matplotlib>=3.5.0",
+    "numpy",  # compatible versions controlled through scipy
+]
```

Comment:
Lower bounds here are set as guesses of reasonable versions, as one should ideally always have lower bounds. These should later be empirically verified for the oldest CPython in `requires-python` (Python 3.7).

Numpy is included without a lower bound as `scipy` intentionally tightly controls the lower and upper bounds of `numpy` versions that are compatible with it and so will shepard `numpy` to the latest compatible release with the installed `scipy`. `numpy` is still included for metadata reasons of specifying all the core dependencies which are imported.